packages/grafana-ui/src/components/Segment/Segment.tsx

```diff
@@ -1,11 +1,12 @@
 import React from 'react';
 import { cx } from 'emotion';
+import _ from 'lodash';
 import { SelectableValue } from '@grafana/data';
 import { SegmentSelect, useExpandableLabel, SegmentProps } from './';
 
 export interface SegmentSyncProps<T> extends SegmentProps<T> {
-  value?: SelectableValue<T>;
-  onChange: (item: SelectableValue<T>) => void;
+  value?: T | SelectableValue<T>;
```

Comment:
This is an awkward change -- how are we supposed to use it?

Reply:
Can we pick one or the other... but not both :)

A similar problem with types -- that should likely be handled elsewhere is with `isMulti` when `isMult` is selected, the actual return value is `Array<SelectableValue<T>>`! so you have to first cast to any to use it!

Reply:
segment does not support isMulti right now, think we should do that with a seperate component that has Array<SelectableValue<T>> type. 

Regarding if value should be the plain value or SelectableValue<T> or not is tricky. Not sure what is right, the current way Select works with only using SelectableValue<T> makes it pretty awkward to use in plain scenarios have you always have to create an "option" value. 

The ant-d SelectValue is even more complex: (we do not use this lib) 
https://github.com/ant-design/ant-design/blob/master/components/select/index.tsx#L52

Reply:
Yes I agree!